kinto/core/storage/__init__.py

```diff
@@ -280,7 +281,25 @@ def purge_deleted(
         """
         raise NotImplementedError
 
-    def get_all(
+    @deprecate_kwargs({"collection_id": "resource_name"})
+    def get_all(self, *args, **kwargs):
+        """Legacy method to support code that relied on the old API where the storage's
+        get_all() would return a tuple of (<list of objects paginated>, <count of all>).
+        Since then, we're being more explicit and expecting the client to deliberately
+        decide if they need a paginated list or a count.
+
+        This method exists solely to make the transition easier.
+        """
+        warnings.warn("Use either self.list_all() or self.count_all()", DeprecationWarning)
+        list_ = self.list_all(*args, **kwargs)
+        kwargs.pop("pagination_rules", None)
+        kwargs.pop("limit", None)
+        kwargs.pop("sorting", None)
+        kwargs.pop("include_deleted", None)
+        count = self.count_all(*args, **kwargs)
```

Comment:
According to this, there should be no call to `.list_all()` and `.count_all()` with the parameter `collection_id`.

Reply:
yep